frontend/javascripts/oxalis/model/sagas/proofread_saga.ts

```diff
@@ -0,0 +1,451 @@
+
+  const volumeTracingId = upToDateVolumeTracing.tracingId;
+  const layerName = volumeTracingId;
+  const serverEditableMapping = yield* call(makeMappingEditable, tracingStoreUrl, volumeTracingId);
+  // The server increments the volume tracing's version by 1 when switching the mapping to an editable one
+  yield* put(setVersionNumberAction(upToDateVolumeTracing.version + 1, "volume", volumeTracingId));
+  yield* put(setMappingNameAction(layerName, serverEditableMapping.mappingName));
+  yield* put(setMappingisEditableAction());
+  yield* put(initializeEditableMappingAction(serverEditableMapping));
+}
+
+function* splitOrMergeAgglomerate(action: MergeTreesAction | DeleteEdgeAction) {
+  const allowUpdate = yield* select((state) => state.tracing.restrictions.allowUpdate);
+  if (!allowUpdate) return;
+
+  const activeTool = yield* select((state) => state.uiInformation.activeTool);
+  if (activeTool !== AnnotationToolEnum.PROOFREAD) return;
+
+  const volumeTracingLayer = yield* select((state) => getActiveSegmentationTracingLayer(state));
+  if (volumeTracingLayer == null) return;
+  const volumeTracing = yield* select((state) => getActiveSegmentationTracing(state));
+  if (volumeTracing == null) return;
+  const { tracingId: volumeTracingId } = volumeTracing;
+
+  const layerName = volumeTracingId;
+  const activeMappingByLayer = yield* select(
+    (state) => state.temporaryConfiguration.activeMappingByLayer,
+  );
+  const mappingInfo = getMappingInfo(activeMappingByLayer, layerName);
+  const { mappingName, mappingType, mappingStatus } = mappingInfo;
+  if (
+    mappingName == null ||
+    mappingType !== "HDF5" ||
+    mappingStatus === MappingStatusEnum.DISABLED
+  ) {
+    Toast.error("An HDF5 mapping needs to be enabled to use the proofreading tool.");
+    return;
+  }
+
+  if (!volumeTracing.mappingIsEditable) {
+    yield* call(createEditableMapping);
+  }
+
+  /* Find out the agglomerate IDs at the two node positions */
+
+  const resolutionInfo = getResolutionInfo(volumeTracingLayer.resolutions);
+  // The mag the agglomerate skeleton corresponds to should be the finest available mag of the volume tracing layer
+  const agglomerateFileMag = resolutionInfo.getLowestResolution();
+  const agglomerateFileZoomstep = resolutionInfo.getLowestResolutionIndex();
+  const { sourceNodeId, targetNodeId } = action;
+
+  const skeletonTracing = yield* select((state) => enforceSkeletonTracing(state.tracing));
+
+  const { trees } = skeletonTracing;
+  const sourceTree = findTreeByNodeId(trees, sourceNodeId).getOrElse(null);
+  const targetTree = findTreeByNodeId(trees, targetNodeId).getOrElse(null);
+
+  if (sourceTree == null || targetTree == null) {
+    return;
+  }
+
+  const sourceNodePosition = sourceTree.nodes.get(sourceNodeId).position;
+  const targetNodePosition = targetTree.nodes.get(targetNodeId).position;
+  const sourceNodeAgglomerateId = yield* call(
+    [api.data, api.data.getDataValue],
```

Comment:
this returns the mapped ids, right?

Reply:
It is ensured that an HDF5 mapping is active, so yes, the mapped data values are returned since the frontend doesn't even know the unmapped ones.